### PR TITLE
add pause logging and a cvar for verbose pause logging (enabled by default)

### LINKED ISSUE #17: Match paused status and durations as log lines output
Currently TF2 servers do not log a (match) pause event in the logs.

Adding one similar to the medic uber duration one (triggered chargedeployed, empty_uber and chargeended (duration 6.2)), might be nice for manual log parsing to get accurate damage per minute calculations.

Most ideal obviously would be to log who triggered the (un)pause, or just make it a world event (`L 08/13/2020 - 21:30:50: World triggered "Pause_Start"`)

Medigun example log lines:
```
L 08/13/2020 - 21:33:30: "Player<1><[U:1:123]><Blue>" triggered "chargedeployed" (medigun "medigun")
L 08/13/2020 - 21:33:33: "Player<1><[U:1:123]><Blue>" triggered "empty_uber"
L 08/13/2020 - 21:33:33: "Player<1><[U:1:123]><Blue>" triggered "chargeended" (duration "3.2")
```
Suggestion for pause log lines:
```
L 08/13/2020 - 21:33:30: "Player<1><[U:1:123]><Blue>" triggered "matchpause"
L 08/13/2020 - 21:40:30: "Player<1><[U:1:123]><Blue>" triggered "matchunpause"
L 08/13/2020 - 21:40:30: World triggered "Pause_Length" (seconds "420.69")
```
The last one is similar to the `L 08/13/2020 - 21:45:30: World triggered "Round_Length" (seconds "91.26")` after a `Round_Win` and new scores get logged.

Additionally at the very end when the `Game_Over` gets triggered and the final scores get logged a total duration of all pauses combined could be outputted to the logs.

Probably should only be the case when log uploads are enabled or have it as a `tftrue_logs_pauses` CVar, similar to the accuracy/includebuffs.

### PR DESCRIPTION
**Pull request by sapphonie**
Closes #17

### CODE REVIEW

File: tournament.cpp [hunk 1]
```diff
@@ -568,29 +576,95 @@ void CTournament::Pause_Callback(ConCommand *pCmd, EDX const CCommand &args)
...
 		}
-		else // Server is not paused
+		// pausing
+		else
 		{
-			g_Plugin.ForwardCommand(pCmd, args);
 			g_Tournament.m_tNextUnpauseAllowed = time(NULL) + tftrue_unpause_delay.GetInt();
-			IClient *pClient = g_pServer->GetClient(*cmd_clientslot);
-			AllMessage(*cmd_clientslot+1, "\x05[TFTrue] The game was paused by \x03%s\x05.\n", pClient->GetClientName());
+			//g_Tournament.m_tLastPauseTime = auto begin = chrono::high_resolution_clock::now();
```
Comment (AnAkkk) [inline tournament.cpp]: Forgot to remove this comment?

File: tournament.h [hunk 2]
```diff
@@ -72,6 +72,7 @@ class CTournament: public IGameEventListener2
 	MapType eMapType = MAPTYPE_UNKNOWN;
 
 	time_t m_tNextUnpauseAllowed = 0;
+	std::chrono::high_resolution_clock::time_point begin;
```
Comment (AnAkkk) [inline tournament.h]: This should be given a better name (it's only for pauses) and follow the naming convention of other member variables.

Comment (sapphonie) [reply tournament.h]: fixed in 8e01600

File: tournament.cpp [hunk 3]
```diff
@@ -568,29 +576,94 @@ void CTournament::Pause_Callback(ConCommand *pCmd, EDX const CCommand &args)
...
+				{
+					V_snprintf(msg, sizeof(msg),
+					  "\"%s<%d><%s><%s>\" triggered \"matchunpause\"\n",
+					  pInfo->GetName(),
+					  pInfo->GetUserID(),
+					  pInfo->GetNetworkIDString(),
+					  szTeamName);
+					engine->LogPrint(msg);
+
+					auto m_tPauseEndTime       = std::chrono::high_resolution_clock::now();
```
Comment (AnAkkk) [inline tournament.cpp]: local variable, so shouldn't start by m_

Comment (sapphonie) [reply tournament.cpp]: You're right, I'm still a newbie at C++ :P

Comment (sapphonie) [reply tournament.cpp]: ac2e44b

**Pull request merged**